google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java

```diff
@@ -204,10 +205,26 @@ public static ClientContext createClientContext(EnhancedBigtableStubSettings set
             ? ((InstantiatingGrpcChannelProvider) builder.getTransportChannelProvider()).toBuilder()
             : null;
 
-    if (builder.getEnableRoutingCookie() && transportProvider != null) {
-      // TODO: this also need to be added to BigtableClientFactory
-      // patch cookies interceptor
-      transportProvider.setInterceptorProvider(() -> ImmutableList.of(new CookiesInterceptor()));
+    ErrorCountPerConnectionMetricTracker errorCountPerConnectionMetricTracker =
+        new ErrorCountPerConnectionMetricTracker(
+            builder.getBackgroundExecutorProvider().getExecutor(),
+            createBuiltinAttributes(builder));
```

Comment:
This should probably be only instantiated if you can set it on the transportProvider.

Also `builder.getBackgroundExecutorProvider().getExecutor()` will have the same problem here as I described in my previous comment...it will return a different instance.

You have 2 options:

1. You can create your own fixed executorProvider (similar to FixedExecutorProvider, but respects close())
2. you can break the circular dependency on the executor by moving it to the start() method of the tracker and call it after the client context is constructed

Reply:
Went with approach 2